app/models/order_item.rb

```diff
@@ -31,4 +35,16 @@ def update_message(level, message)
     progress_messages << ProgressMessage.new(:level => level, :message => message, :tenant_id => self.tenant_id)
     touch
   end
+
+  private
+
+  def sanitize_parameters
+    # Store the API accessible parameters with the protected field values masked
+    self.service_parameters_raw = self[:service_parameters]
+    self[:service_parameters] = Catalog::OrderItemSanitizedParameters.new(:order_item => self).process.sanitized_parameters
+  end
+
+  def service_parameters_raw=(val)
+    self[:service_parameters_raw] = val
```

Comment:
You can probably just inline this on line 43 since the method is private and we're not calling it anywhere else.

Reply:
Ended up adding more logic to this method.  The other option here is to move the `contains_sanitized_values?` check in `sanitize_parameters` and make `service_parameters_raw=` as readonly, but I really wanted this setter method to be completely private to the object.  It will be obvious is someone tries to call it this way, as opposed to using `attr_readonly` which would silently ignore the call.